src/storage/txn/actions/acquire_pessimistic_lock.rs

```diff
@@ -73,10 +114,17 @@ pub fn acquire_pessimistic_lock<S: Snapshot>(
                 .acquire_pessimistic_lock
                 .inc();
         }
-        return Ok(val);
+        return Ok((val, old_value));
     }
 
+    // Following seek_write read the previous write.
+    let (prev_write_loaded, mut prev_write) = (true, None);
     if let Some((commit_ts, write)) = reader.seek_write(&key, TimeStamp::max())? {
+        // Find a previous write.
+        if need_old_value {
+            prev_write = Some(write.clone());
+        }
```

Comment:
There should be some way to avoid the cloning by moving `write` to `prev_write` and use `prev_write` instead of `write` in the if block. But it's ok to me since the current code looks clear.

Reply:
It is possible to avoid clone when need_value is false, though it may make code more complicated.